pkg/aws/eni/limits/limits.go

```diff
@@ -7,847 +7,108 @@ package limits
-		"x2iedn.xlarge":     {Adapters: 4, IPv4: 15, IPv6: 15, HypervisorType: "nitro"},
-		"x2iezn.12xlarge":   {Adapters: 15, IPv4: 50, IPv6: 50, HypervisorType: "nitro"},
-		"x2iezn.2xlarge":    {Adapters: 4, IPv4: 15, IPv6: 15, HypervisorType: "nitro"},
-		"x2iezn.4xlarge":    {Adapters: 8, IPv4: 30, IPv6: 30, HypervisorType: "nitro"},
-		"x2iezn.6xlarge":    {Adapters: 8, IPv4: 30, IPv6: 30, HypervisorType: "nitro"},
-		"x2iezn.8xlarge":    {Adapters: 8, IPv4: 30, IPv6: 30, HypervisorType: "nitro"},
-		"x2iezn.metal":      {Adapters: 15, IPv4: 50, IPv6: 50, HypervisorType: ""},
-		"z1d.12xlarge":      {Adapters: 15, IPv4: 50, IPv6: 50, HypervisorType: "nitro"},
-		"z1d.2xlarge":       {Adapters: 4, IPv4: 15, IPv6: 15, HypervisorType: "nitro"},
-		"z1d.3xlarge":       {Adapters: 8, IPv4: 30, IPv6: 30, HypervisorType: "nitro"},
-		"z1d.6xlarge":       {Adapters: 8, IPv4: 30, IPv6: 30, HypervisorType: "nitro"},
-		"z1d.large":         {Adapters: 3, IPv4: 10, IPv6: 10, HypervisorType: "nitro"},
-		"z1d.metal":         {Adapters: 15, IPv4: 50, IPv6: 50, HypervisorType: ""},
-		"z1d.xlarge":        {Adapters: 4, IPv4: 15, IPv6: 15, HypervisorType: "nitro"},
+func NewLimitsGetter(api ec2API, triggerMinInterval, ec2APITimeout time.Duration, ec2APIRetryCount int) (*LimitsGetter, error) {
+	l := &LimitsGetter{
+		triggerDone: make(chan struct{}),
+		m:           make(map[string]ipamTypes.Limits),
+	}
+	err := l.initEC2APIUpdateTrigger(api, triggerMinInterval, ec2APITimeout, ec2APIRetryCount)
+	if err != nil {
+		log.WithError(err).Warning("Unable to initialize EC2 API update trigger")
+		return nil, err
 	}
+	return l, nil
 }
 
 // Get returns the instance limits of a particular instance type.
-func Get(instanceType string) (limit ipamTypes.Limits, ok bool) {
-	limitsOnce.Do(populateStaticENILimits)
-
-	limits.RLock()
-	limit, ok = limits.m[instanceType]
-	limits.RUnlock()
-	return
-}
-
-// UpdateFromUserDefinedMappings updates limits from the given map.
-func UpdateFromUserDefinedMappings(m map[string]string) (err error) {
-	limitsOnce.Do(populateStaticENILimits)
-
-	limits.Lock()
-	defer limits.Unlock()
-
-	for instanceType, limitString := range m {
-		limit, err := parseLimitString(limitString)
-		if err != nil {
-			return err
-		}
-		// Add or overwrite limits
-		limits.m[instanceType] = limit
+func (l *LimitsGetter) Get(instanceType string) (ipamTypes.Limits, bool) {
+
+	l.RLock()
+	if limit, ok := l.m[instanceType]; ok {
+		l.RUnlock()
+		log.WithFields(logrus.Fields{
+			"limit":        limit,
+			"instanceType": instanceType,
+		}).Debug("Get limits from cache")
+		return limit, true
+	}
+	minInterval := l.triggerMinInterval
+	ec2apiTimeoutWithJitter := l.ec2APITimeout + time.Duration(float64(l.ec2APITimeout)*0.1)
+	lastUpdate := l.lastUpdate
+	currentDone := l.triggerDone
+	l.RUnlock()
+	// If the update happened less than minInterval ago and Get() is called again within internal
```

Comment:
this looks like a typo:

```suggestion
	// If the update happened less than minInterval ago and Get() is called again within the interval
```